### DISCUSSION TITLE
readonly property in abstract base class

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3231] [Questions]
I just found out readonly property needs to have protected accessor in abstract base class. Every day I learn something new, but just wanted confirm, is this the case?

Kind regards

**Comment by rockfordlhotka** [comment]
Properties do need to have some sort of setter, if that's what you are asking?

**Reply by Chicagoan2016** [reply]
@rockfordlhotka just wondering if 'protected' setter is the way to go **OR** do have a Csla type declaration to be used in base classes?
(e.g. On page 18 of 'Creating busines objects' we have various Property Declarations Csla provides i.e. readonly, intergraph, read-write etc.)

Regards
